src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionposthearingapplication/ActionPostHearingApplicationAboutToSubmitHandlerTest.java

```diff
@@ -0,0 +1,103 @@
+import static uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.EventType.ACTION_POST_HEARING_APPLICATION;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.HearingRoute.GAPS;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.HearingRoute.LIST_ASSIST;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DocumentGeneration;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DocumentLink;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DocumentStaging;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SchedulingAndListingFields;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ActionPostHearingApplicationAboutToSubmitHandlerTest {
+
+    private static final String DOCUMENT_URL = "dm-store/documents/123";
+
+    private static final String USER_AUTHORISATION = "Bearer token";
+
+    @InjectMocks
+    private ActionPostHearingApplicationAboutToSubmitHandler handler;
+
+    @Mock
+    private Callback<SscsCaseData> callback;
+
+    @Mock
+    private CaseDetails<SscsCaseData> caseDetails;
+
+    private SscsCaseData sscsCaseData;
+
+    @Before
+    public void setUp() {
+        ReflectionTestUtils.setField(handler, "isScheduleListingEnabled", true);
+
+        when(callback.getEvent()).thenReturn(ACTION_POST_HEARING_APPLICATION);
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+
+        sscsCaseData = SscsCaseData.builder()
+            .schedulingAndListingFields(SchedulingAndListingFields.builder()
+                .hearingRoute(LIST_ASSIST).build())
+            .ccdCaseId("ccdId")
+            .documentGeneration(DocumentGeneration.builder()
+                .directionNoticeContent("Body Content")
+                .build())
+            .documentStaging(DocumentStaging.builder()
+                .previewDocument(DocumentLink.builder()
+                    .documentUrl(DOCUMENT_URL)
+                    .documentBinaryUrl(DOCUMENT_URL + "/binary")
+                    .documentFilename("decisionIssued.pdf")
+                    .build())
+                .build())
+            .build();
+
+        when(caseDetails.getCaseData()).thenReturn(sscsCaseData);
+    }
+
+    @Test
+    public void givenAValidAboutToSubmitEvent_thenReturnTrue() {
+        assertThat(handler.canHandle(ABOUT_TO_SUBMIT, callback)).isTrue();
+    }
+
+    @Test
+    public void givenSchedulingAndListingEnabledFalse_thenReturnFalse() {
+        ReflectionTestUtils.setField(handler, "isScheduleListingEnabled", false);
```

Comment:
Prefer to use constructors to avoid needing to do that

Reply:
I prefer the opposite really, the constructors can get really messy, this I find a lot more clean

Reply:
Yes, the constructor injection is being used for injecting the other beans, however, the field injection is only for the primitive boolean from the application.yml. Reflection is a perfectly common tool used for unit tests when injecting values for unit tests that come from application.yml.

Reply:
Using ReflectionTestUtils hides what's actually happening - underneath the hood it is modifying the accessibility of fields at runtime which can be very dangerous, even in tests. Whilst it might be a bit more verbose, using constructors makes it very clear what the object is to be instantiated with. In this case it might be simple enough, but using Reflection is something that should be avoided unless absolutely required.

Reply:
I have added them to the constructor :)